persistence/src/main/java/edu/unc/lib/dl/persist/services/destroy/DestroyObjectsJob.java

```diff
@@ -0,0 +1,154 @@
+           }
+        } catch (Exception e) {
+             tx.cancel(e);
+        } finally {
+             tx.close();
+        }
+    }
+
+    private void destroyTree(RepositoryObject rootOfTree) throws FedoraException {
+        if (rootOfTree instanceof ContentContainerObject) {
+            ContentContainerObject container = (ContentContainerObject) rootOfTree;
+            List<ContentObject> members = container.getMembers();
+            for (ContentObject member : members) {
+                destroyTree(member);
+            }
+        } else if (rootOfTree instanceof FileObject) {
+            FileObject file = (FileObject) rootOfTree;
+            List<BinaryObject> binaries = file.getBinaryObjects();
+            for (BinaryObject binary : binaries) {
+                destroyTree(binary);
+            }
+        }
+        // destroy root of tree
+        Model stoneModel = rootOfTree.getModel();
+        stoneModel = convertModelToTombstone(rootOfTree);
+        repoObjFactory.createObject(rootOfTree.getUri(), stoneModel);
+
+        //add premis event to tombstone
+        rootOfTree.getPremisLog().buildEvent(Premis.Deletion)
+        .addImplementorAgent(agent.getUsernameUri())
+        .addEventDetail("Item deleted from repository and replaced by tombstone")
+        .write();
+
+        // persist model
+        repoObjFactory.createObject(rootOfTree.getUri(), stoneModel);
+    }
+
+    private Model convertModelToTombstone(RepositoryObject destroyedObj) {
+        Model model = destroyedObj.getModel();
+        Resource resc = destroyedObj.getResource();
+
+        if (destroyedObj instanceof BinaryObject) {
+            // whitelisted properties for tombstone model of binaries
+            Selector selectEbucoreFilename = new SimpleSelector(resc, Ebucore.filename, (Object) null);
+            Selector selectDcFilename = new SimpleSelector(resc, DcElements.title, (Object) null);
+            Selector selectPremisOriginalName = new SimpleSelector(resc, Premis.hasOriginalName, (Object) null);
+            Selector selectMimetype = new SimpleSelector(resc, Ebucore.hasMimeType, (Object) null);
+            Selector selectChecksum= new SimpleSelector(resc, Premis.hasMessageDigest, (Object) null);
+            Selector selectFilesize = new SimpleSelector(resc, Premis.hasSize, (Long) null);
+
+            StmtIterator iter = model.listStatements();
+            while (iter.hasNext()) {
+                Statement s = iter.nextStatement();
+                if (!(selectEbucoreFilename.test(s) || selectDcFilename.test(s) || selectPremisOriginalName.test(s)
+                        || selectMimetype.test(s) || selectChecksum.test(s) || selectFilesize.test(s))) {
+                    model.remove(s);
+                }
+            }
+        } else {
+            // retain only the object-type of a non-binary object
+            Selector selectRdfType = new SimpleSelector(resc, RDF.type, (Object) null);
```

Comment:
We also want to keep the dc:title and premis:hasOriginalFilename here too (imported folders can have an original filename)